.github/workflows/main.yml

```diff
@@ -39,23 +39,42 @@ jobs:
           python -m pip install --upgrade pip
           pip install -r requirements-dev.txt
 
-      - name: Run pycodestyle
-        run: |
-          pycodestyle api/*.py
-
-      - name: Run pylint
-        run: |
-          pylint api.auth
-          pylint api.db
-          pylint -d R0903 api.models
-          pylint api.pubsub
-          pylint test
-          pylint api.main
-
       - name: Export environment variables
         run: |
           echo "SECRET_KEY=$(openssl rand -hex 32)" >> $GITHUB_ENV
 
       - name: Run pytest
         run: |
           pytest -v test/
+
+  lint:
+    runs-on: ubuntu-22.04
+    name: Lint
+
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # This is necessary to get the commits
+
+      - name: Set up Python environment
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+
+      - name: Get changed python files between base and head
+        run: >
+          echo "CHANGED_FILES=$(echo $(git diff --name-only ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} -- | grep \.py))" >> $GITHUB_ENV
+
+      - if: env.CHANGED_FILES
+        uses: marian-code/python-lint-annotate@v3
+        with:
+          python-root-list: ${{ env.CHANGED_FILES }}
+          use-black: false
+          use-flake8: false
+          use-isort: false
+          use-mypy: false
+          use-pycodestyle: true
+          use-pydocstyle: false
+          use-vulture: false
+          python-version: "3.10"
```

Comment:
I believe this should be dropped as the setup-python step already sets it up with a single version.  IIRC this option is only useful when several Python versions are installed.

Reply:
setup-python step explicitly declares python version 3.10 to be used.
But somehow the lint job uses python version 3.8 from actions within `marian-code/python-lint-annotate@v3`.
I added `python-version` to correct it.

Reply:
Ah I guess `ubuntu-22.04` comes with Python 3.8 already installed then.  In this case I guess we need to keep the explicit `python-version` option and I should do the same with `kernelci-core`.

Reply:
Yes.

Reply:
Actually, I double checked and `ubuntu-22.04` comes with Python 3.10.  The `marian-code` action runs its own `setup-python` step with 3.8 by default:
```
Run actions/setup-python@v2
  with:
    python-version: 3.8
    architecture: x64
    cache: pip
    token: ***
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.10.10/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.10/x64/lib
    CHANGED_FILES: kci kernelci/config/api.py
```
So we can just drop the first `setup-python` step in `main.yml` and rely only on the `python-version` in the annotation action.

Reply:
Okay. Thanks for confirming.

Reply:
Also, why the value 32 is used in `fetch-depth: 32` in `kernelci-core`?

Reply:
Because we never have PRs with more than 32 commits, in practice.

Reply:
I've added a commit to this PR to use the `python-version` attribute in the annotation action:
https://github.com/kernelci/kernelci-core/pull/1748